lib/cretonne/src/dominator_tree.rs

```diff
@@ -323,6 +329,72 @@ impl DominatorTree {
     }
 }
 
+impl DominatorTree {
+    /// When splitting an `Ebb` using `Layout::split_ebb`, you can use this method to update
+    /// the dominator tree locally rather than recomputing it.
+    ///
+    /// `old_ebb` is the `Ebb` before splitting, and `new_ebb` is the `Ebb` which now contains
+    /// the second half of `old_ebb`.
+    pub fn recompute_split_ebb(&mut self, old_ebb: Ebb, new_ebb: Ebb, split_jump_inst: Inst) {
```

Comment:
Since this API is a bit tricky, I think it would be a good idea to add a couple `debug_assert!`s here: `split_jump_inst` is the terminator of `old_ebb`, and it is a jump to `new_ebb`.